src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -192,5 +206,37 @@ public static bool NTLM_Available()
             return GetConditionValue(nameof(NTLM_Available),
                                      Server_Is_LocalHost);
         }
+
+        // Returns the Domain if available.
+        // TestProperties takes precedence, but if it has not been specified
+        // and this is a Windows client, we infer it.
+        public static string GetDomain()
+        {
+            return ConditionalTestDetectors.GetDomain();
+        }
+
+        // Returns the explicit user name if available
+        public static string GetExplicitUserName()
+        {
+            return ConditionalTestDetectors.GetExplicitUserName();
+        }
+
+        // Returns the explicit password if available
+        public static string GetExplicitPassword()
+        {
+            return ConditionalTestDetectors.GetExplicitPassword();
+        }
+
+        // Gets the UPN if available
+        public static string GetUPN()
```

Comment:
Reviewers: one observation about the changes -- before these changes a test using ConditionalFact like UPN_Available had to just magically *know* which TestProperty to use for the value, which was confusing.  With these changes, the condition *and* value checked by the condition are both here in the base class.  Therefore a test testing base.UPN_Available() can just use base.GetUPN() to get the value associated with that condition.